internal/core/window.rs

```diff
@@ -681,19 +681,15 @@ impl Window {
         self.scale_factor.as_ref().set(factor)
     }
 
-    /// Returns the font properties that are set on the root item if it's a Window item.
-    pub fn default_font_properties(&self) -> crate::graphics::FontRequest {
-        self.try_component()
-            .and_then(|component_rc| {
-                let component = ComponentRc::borrow_pin(&component_rc);
-                let root_item = component.as_ref().get_item_ref(0);
-                ItemRef::downcast_pin(root_item).map(
-                    |window_item: Pin<&crate::items::WindowItem>| {
-                        window_item.default_font_properties()
-                    },
-                )
-            })
-            .unwrap_or_default()
+    /// Returns the window item that is the first item in the component.
+    pub fn window_item(&self) -> Option<ItemRc> {
```

Comment:
Is there a way to return `Pin<&crate::items::WindowItem>` Or a `VRcMapped<..., crate::items::WindowItem>`
or maybe do a `with_window_item<R>(_: impl FnOnce(Pin<&crate::items::WindowItem>`) ->R) ->Option<R>`

Because I think we use that a lot, but this function isn't so usefull if it doesn't do the downcast.

Reply:
Yeah, I'd like that, too - but failed due to the temporaries involved. `VRCMapped` is a fascinating idea - I'll try that first and if it doesn't work I'll go for `with_window_item`.